staywoke/lambdas
# Standards

#### What's this PR do?

Initial project tooling to add linting, unit tests & code coverage reports to repository.  It's also the first required step before I can move on to integrating Circle CI to the master branch.  So this PR is not so much about the lambda functions themselves, but more around the architecture that tests the lambda function.

**Goals of this PR:**

- [X] Add `.editorconfig` file IDE's that support it
- [X] Add `.github` folder with templates for code contribution
- [X] Add foundation for documenting lambda functions
- [X] Setup Unit Testing
- [X] Setup Code Coverage

#### Where should the reviewer start?

Clone the repo, run `npm install` and then fire off the unit tests via `npm test`.  You might also wish to test individual NPM scripts manually, so check out the package.json file for those.

#### How should this be manually tested?

```bash
npm install
npm test
```

You will likely want to test the deployment scripts, which can be done via:

```
npm run deploy:production
```

This is not stitched up to production yet, but you will see a a console message showing you that the packages are being zipped up for release, and they should be available in `./dist` folder.

#### Screenshots (if appropriate)

> `npm test`

![screenshot 2018-03-26 at 1 22 26 pm](https://user-images.githubusercontent.com/508411/37921804-ca2ba376-30f8-11e8-9f3d-6f40ee448525.png)

> `npm run deploy:production`

![screenshot 2018-03-27 at 12 26 02 am](https://user-images.githubusercontent.com/508411/37947016-b05fd4c2-3156-11e8-9913-ed0dee4f06af.png)

#### What gif best describes this PR or how it makes you feel?

![refreshing](https://user-images.githubusercontent.com/508411/37922018-747c0992-30f9-11e8-867c-2c46b91e9c61.gif)

#### Definition of Done:

- [X] You have run `npm test` and it passes
- [X] You have added code comments to all code being submitted
